recipes/libsodium/all/conanfile.py

```diff
@@ -103,73 +142,100 @@ def _msvc_sln_folder(self):
 
         return folder.get(str(self.settings.compiler.version))
```

Comment:
The checks above should be using `is_msvc` id image?

Reply:
I don't know what you mean by "id image" ?

`_msvc_sln_folder()` is only called by `_build_msvc()`, which is only called within `build()`, and that call is protected by `is_msvc()`,  so normally I might put in a preassert to check is_msvc() and throw a logic_error(), if I thought there was a chance I'd call it incorrectly ...

How do you think it should be changed?
Maybe `else if settings.compiler == "msvc":`
and then `else: raise HowOnEarthDidWeReachThisLine()`
?

Reply:
Auto correct since I work on my phone like a monster 😈 

I this you should replace the `if self.settings.compiler == "Visual Studio"` with `if is_msvc(self)` to help future proof the recipe

Reply:
Ohhh... I just got it. The `else` is taking care of "msvc" that was totally escaping me.

Maybe just add a comment for future me 🙏

Reply:
I add an elif "msvc" and raised an exception if the compiler is different.
As you say, future proofing, and it is more clear to me too.